kernel mode: Mark new veth peer as up

cathay4t opened the pull request:
The nispor 1.2.28 will not automatically mark new veth peer up now.
We should do it in nmstate kernel mode.

The test case `test_add_and_remove_veth_kernel_mode` is enough to
reproduce the issue and confirm the fix.

jvaclav-rh commented on rust/src/lib/nispor/apply.rs:
```diff
@@ -42,6 +42,9 @@ pub(crate) async fn nispor_apply(
             && i.for_apply.as_ref().is_some()
     }) {
         np_ifaces.push(nmstate_iface_to_np(merged_iface)?);
+        if let Some(np_iface) = process_new_veth_peer(merged_iface) {
```
Could it happen that the other veth peer is already defined in `merged_state.interfaces`? If so, should this call be skipped?

cathay4t replied:
Yeah. Let me skip it.

cathay4t replied:
Done.

ihuguet commented on rust/src/lib/nispor/veth.rs:
```diff
@@ -35,3 +38,33 @@ pub(crate) fn nms_veth_conf_to_np(
+            if let Some(peer) =
+                apply_iface.veth.as_ref().map(|v| v.peer.as_str())
+            {
+                // Mark new veth peer as up if not desired
```
if not ~desired~ mentioned in desired state. That would be clearer.

Apart from that, the code LGTM, but I'd add a test case if possible.

cathay4t replied:
Test case `test_add_and_remove_veth_kernel_mode` is exactly the test case.

Comment updated.

cathay4t committed:
kernel mode: Mark new veth peer as up

The nispor 1.2.28 will not automatically mark new veth peer up now.
We should do it in nmstate kernel mode.

The test case `test_add_and_remove_veth_kernel_mode` is enough to
reproduce the issue and confirm the fix.

Signed-off-by: Gris Ge <fge@redhat.com>
